### PR TITLE
Make global scope enums accessible as types in GDScript

### LINKED ISSUE #72963: `_parse_property` in `EditorInspectorPlugin` is asking for `::Type` instead of `int` in `type` parameter.
### Godot version

4.0 rc1

### System information

Manjaro Linux using i3, Foward+

### Issue description

All of my `EditorInspectorPlugin`s are broken with the following error:
```
Line 34:The function signature doesn't match the parent. Parent signature is "_parse_property(Object, ::Type, String, int, String, int, bool) -> bool".
```

I tried to change it from `int` to `Type` like it shows in the documentation and it also doesn't work:
image
image

And keeping it as an `int` just gives me the function signature mismatch error above.

### Steps to reproduce

Create a new `EditorInspectorPlugin` script and try to override the virtual method `_parse_property`

### Minimal reproduction project

See Steps to reproduce above.

### LINKED ISSUE #73046: RC1: Using engine `GlobalScript` enum  as return type ends up in a compile error
### Godot version

v4.0.rc1.official [8843d9ad3]

### System information

ALL

### Issue description

I build a mock/spy and double all engine functions.
To double successfully i have to "shadow" the core function with the same function signature.

Since Godot RC1 it needs to using the correct enum as return type otherwise it will end up in a error.
`The function signature doesn't match the parent. Parent signature is "get_texture_filter() -> CanvasItem::TextureFilter".`

Godot RC1
```gdscript
@warning_ignore("native_method_override")
func get_texture_filter() -> TextureFilter:
  ...
```

Godot betaXX worked with an int 
```gdscript
@warning_ignore("native_method_override")
func get_texture_filter() -> int:
```

---
So i addapt the doubler to use the right enum when the method descriptor has defined a enum as return.
But it fails for the enum`@GlobalScope:Error` 

```gdscript
@warning_ignore("native_method_override")
func connect(signal_, callable_, flags_=0) -> Error:
```
`Parser Error: "Error" was not found in the current scope.`

### Steps to reproduce

Use this simple part to reproduce

```gdscript
extends RefCounted


# using an existing engine enum -- WORKS
func foo() -> CanvasItem.TextureFilter:
	return CanvasItem.TEXTURE_FILTER_PARENT_NODE


# using GlobalScript enum fails -- FAILS
func bar() -> Error:
	return OK

```

### Minimal reproduction project

N/A

### PR DESCRIPTION
**Pull request by vnen**
Add functions to CoreConstant so enums can be properly deduced. Also add the enums in release builds to make consistent with ClassDB enums and avoid differences in script compilation between debug and release.

Fix #72963
Fix #73046
Closes #72984 (superseded)

### CODE REVIEW

File: modules/gdscript/gdscript_analyzer.cpp [hunk 1]
```diff
@@ -3861,7 +3947,12 @@ void GDScriptAnalyzer::reduce_subscript(GDScriptParser::SubscriptNode *p_subscri
 			}
 		}
 		if (!valid) {
-			push_error(vformat(R"(Cannot find member "%s" in base "%s".)", p_subscript->attribute->name, type_from_metatype(base_type).to_string()), p_subscript->attribute);
+			GDScriptParser::DataType attr_type = p_subscript->attribute->get_datatype();
+			if (!p_can_be_pseudo_type && (attr_type.is_pseudo_type || result_type.is_pseudo_type)) {
+				push_error(vformat(R"(Type "%s" in base "%s" cannot be used on its own.)", p_subscript->attribute->name, type_from_metatype(base_type).to_string()), p_subscript->attribute);
```
Comment (anvilfolk) [inline modules/gdscript/gdscript_analyzer.cpp:3952]: Doing a super quick read of the PR, but this feels like a pretty cryptic error message. Is there maybe a suggestion that can be given to the user? Like asking if they were looking for a subtype, or for an enum value, or something like that?

**Review by akien-mga — APPROVED** [review]

File: modules/gdscript/gdscript_analyzer.cpp (hunk 1 above)
Comment (vnen) [reply modules/gdscript/gdscript_analyzer.cpp:3952]: Sure, we can always improve the error messages.

**Comment by akien-mga** [comment]
Thanks!

**Comment by MikeSchulze** [comment]
👍

**Pull request merged**